lib/lotus/web/controllers/export_controller.ex

```diff
@@ -0,0 +1,106 @@
+defmodule Lotus.Web.ExportController do
+  @moduledoc """
+  Controller for streaming CSV exports.
+  """
+
+  use Phoenix.Controller, formats: [:csv]
+
+  alias Lotus.Export
+
+  plug(:fetch_query_params)
+
+  @max_age 300
+
+  @doc """
+  Streams a CSV export directly to the response.
+
+  Expects a signed token containing:
+  - query_id: ID of the query to export
+  - repo: data repository name
+  - vars: variable values
+  - search_path: optional search path
+  - filename: suggested filename for download
+  """
+  def csv(conn, %{"token" => token}) do
+    endpoint = Phoenix.Controller.endpoint_module(conn)
+
+    case Phoenix.Token.verify(endpoint, salt(endpoint), token, max_age: @max_age) do
+      {:ok, export_params} ->
+        stream_csv_export(conn, export_params)
+
+      {:error, :expired} ->
+        conn
+        |> put_status(:unauthorized)
+        |> text("Export link has expired. Please try again")
+
+      {:error, _reason} ->
+        conn
+        |> put_status(:unauthorized)
+        |> text("Invalid export token")
+    end
+  end
+
+  def csv(conn, _params) do
+    conn
+    |> put_status(:bad_request)
+    |> text("Missing export token.")
+  end
+
+  defp stream_csv_export(conn, export_params) do
+    query = build_query(export_params)
+    filename = Map.get(export_params, "filename", "export.csv")
+
+    conn
+    |> put_resp_content_type("text/csv")
+    |> put_resp_header("content-disposition", ~s(attachment; filename="#{filename}"))
+    |> send_chunked(200)
+    |> stream_chunks(query, export_params)
+  end
+
+  defp build_query(%{"query_id" => query_id}) when not is_nil(query_id) do
+    Lotus.get_query(query_id)
+  end
+
+  defp build_query(_), do: nil
+
+  defp stream_chunks(conn, nil, _export_params) do
+    {:ok, conn} = chunk(conn, "Error: Query not found")
```

Comment:
As mentioned before, I don't think these error messages inside the chunked response are going to be read by anyone. I'd prefer if the controller simply returned a 404 (+ text response) when the query can't be found.